modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java

```diff
@@ -155,6 +155,16 @@ public Object visitComplexPropertyExpression(
 						relatedObjectDefinitionId));
 		}
 
+		if (propertyExpression instanceof ComplexPropertyExpression) {
+			return _handleComplexPropertyExpressionPropertyExpression(
+				propertyExpression,
+				new ArrayList<String>() {
```

Comment:
What about:

```
Arrays.asList(complexPropertyExpression.getName())
```

Reply:
I tried but seems like it creates an immutable list and I cannot add objects to it inside the method using `add` (I think there are other ways to add it)
Yes, makes no sense, I will try again just in case I did something wrong when I tested it, but that's the reason why I have to put the `new ArrayList`

Reply:
I leave it at it is 🤷 Let's see what Brian says